src/files.h

```diff
@@ -0,0 +1,31 @@
+#include <stdbool.h>
+#include <stdint.h>
+
+struct x16file;
+
+#define XSEEK_SET	0
+#define XSEEK_END	1
+#define XSEEK_CUR	2
+
+bool file_is_compressed_type(char const *path);
+
+void files_shutdown();
+
+struct x16file *x16open(const char *path, const char *attribs);
+void x16close(struct x16file *f);
+
+int64_t x16size(struct x16file *f);
+int x16seek(struct x16file *f, int64_t pos, int origin);
+int64_t x16tell(struct x16file *f);
+
+
+int x16write8(struct x16file *f, uint8_t val);
+int x16write16(struct x16file *f, uint16_t val);
+int x16write32(struct x16file *f, uint32_t val);
+
+uint8_t x16read8(struct x16file *f);
+uint16_t x16read16(struct x16file *f);
+uint32_t x16read32(struct x16file *f);
```

Comment:
What do you have in mind for these read and write value functions? I'm wondering what utility they have w/ host-endian-specific behaviors.

Reply:
Hmm. I hadn't considered endian-ness, I should pull out the -16 and -32 bit flavors, I don't think there were any use cases for those in the rest of the code anyways, and other places that use similar-looking SDL functions were only doing read8.